boa_engine/src/object/internal_methods/immutable.rs

```diff
@@ -0,0 +1,39 @@
+use crate::{
+    object::{JsObject, JsPrototype},
+    Context, JsResult,
+};
+
+use super::{InternalObjectMethods, ORDINARY_INTERNAL_METHODS};
+
+/// Definitions of the internal object methods for [**Module Namespace Exotic Objects**][spec].
+///
+/// [spec]: https://tc39.es/ecma262/#sec-module-namespace-exotic-objects
+pub(crate) static IMMUTABLE_PROTOTYPE_EXOTIC_INTERNAL_METHODS: InternalObjectMethods =
+    InternalObjectMethods {
+        __set_prototype_of__: set_prototype_of,
+        ..ORDINARY_INTERNAL_METHODS
+    };
+
+/// [`[[SetPrototypeOf]] ( V )`][spec].
+///
+/// [spec]: https://tc39.es/ecma262/#sec-immutable-prototype-exotic-objects-setprototypeof-v
+#[allow(clippy::needless_pass_by_value)]
+pub(crate) fn set_prototype_of(
```

Comment:
We should probably prefix with `immutable_prototype_exotic_`

```suggestion
pub(crate) fn immutable_prototype_exotic_set_prototype_of(
```

To be consistent with other internal methods implementations